connector/src/main/java/com/datastax/oss/pulsar/source/converters/NativeJsonConverter.java

```diff
@@ -0,0 +1,407 @@
+            case ProtocolConstants.DataType.SMALLINT:
+                return jsonNodeFactory.numberNode((int) record.getShort(identifier));
+            case ProtocolConstants.DataType.INT:
+                return jsonNodeFactory.numberNode(record.getInt(identifier));
+            case ProtocolConstants.DataType.BIGINT:
+                return jsonNodeFactory.numberNode(record.getLong(identifier));
+            case ProtocolConstants.DataType.INET:
+                return jsonNodeFactory.textNode(record.getInetAddress(identifier).getHostAddress());
+            case ProtocolConstants.DataType.DOUBLE:
+                return jsonNodeFactory.numberNode(record.getDouble(identifier));
+            case ProtocolConstants.DataType.FLOAT:
+                return jsonNodeFactory.numberNode(record.getFloat(identifier));
+            case ProtocolConstants.DataType.BOOLEAN:
+                return jsonNodeFactory.booleanNode(record.getBoolean(identifier));
+            case ProtocolConstants.DataType.TIMESTAMP:
+                return jsonNodeFactory.numberNode(record.getInstant(identifier).toEpochMilli());
+            case ProtocolConstants.DataType.DATE: // Mimic Avro date (epoch days)
+                return jsonNodeFactory.numberNode((int) record.getLocalDate(identifier).toEpochDay());
+            case ProtocolConstants.DataType.TIME: // Mimic Avro time (microseconds)
+                return jsonNodeFactory.numberNode (record.getLocalTime(identifier).toNanoOfDay() / 1000);
+            case ProtocolConstants.DataType.BLOB:
+                return jsonNodeFactory.binaryNode(getBytes(record.getByteBuffer(identifier)));
+            case ProtocolConstants.DataType.UDT:
+                return buildUDTValue(record.getUdtValue(identifier));
+            case ProtocolConstants.DataType.DURATION:
+                return cqlDurationToJsonNode(record.getCqlDuration(identifier));
+            case ProtocolConstants.DataType.DECIMAL:
+                return bigDecimalToJsonNode(record.getBigDecimal(identifier));
+            case ProtocolConstants.DataType.VARINT:
+                Conversion<BigInteger> conversion = SpecificData.get().getConversionByClass(BigInteger.class);
+                ByteBuffer bf = conversion.toBytes(record.getBigInteger(identifier), CqlLogicalTypes.varintType, CqlLogicalTypes.CQL_VARINT_LOGICAL_TYPE);
+                return jsonNodeFactory.binaryNode(getBytes(bf));
+            case ProtocolConstants.DataType.LIST: {
+                ListType listType = (ListType) dataType;
+                String path = getSubSchemaPath(record, identifier);
+                org.apache.avro.Schema listSchema = subSchemas.get(path);
+                List listValue = record.getList(identifier, CodecRegistry.DEFAULT.codecFor(listType.getElementType()).getJavaType().getRawType());
+                log.info("field={} listSchema={} listValue={}", identifier, listSchema, listValue);
```

Comment:
if (log.isDebugEnabled()...log.debug

here we are in a hotpath, we are going to write millions of log lines

Reply:
good catch! fixing...